src/System.Text.Primitives/System/Text/Encoders/UnicodeScalar.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Globalization;
+
+namespace System.Buffers.Text
+{
+    /// <summary>
+    /// Represents a 24-bit Unicode scalar value.
+    /// A scalar value is any value in the range [U+0000..U+D7FF] or [U+E000..U+10FFFF].
+    /// </summary>
+    public struct UnicodeScalar : IComparable<UnicodeScalar>, IEquatable<UnicodeScalar>
+    {
+        /// <summary>
+        /// The Unicode Replacement Character U+FFFD.
+        /// </summary>
+        public static readonly UnicodeScalar ReplacementChar = new UnicodeScalar(0xFFFD);
+
+        /// <summary>
+        /// The integer value of this scalar.
+        /// </summary>
+        public readonly int Value; // = U+0000 if using default init
+
+        /// <summary>
+        /// Constructs a Unicode scalar from the given UTF-16 code point.
+        /// The code point must not be a surrogate.
+        /// </summary>
+        /// <param name="char"></param>
+        public UnicodeScalar(char @char)
```

Comment:
What is the meaning behind the '@' before the char parameter?
**Edit:** Interesting... using reserved keywords as parameter name - https://stackoverflow.com/questions/1038674/c-sharp-prefixing-parameter-names-with